tests/ops/op_math/test_controlled_ops.py

```diff
@@ -428,3 +441,140 @@ def test_unitary_check(self):
             qml.ControlledQubitUnitary(
                 not_unitary, control_wires=[0, 2], wires=1, unitary_check=True
             )
+
+
+class TestOperations:
+    @pytest.mark.parametrize("op_cls, _", NON_PARAMETRIZED_OPERATIONS)
+    def test_nonparametrized_op_copy(self, op_cls, _, tol):
+        """Tests that copied nonparametrized ops function as expected"""
+        op = op_cls(wires=0 if op_cls.num_wires is AnyWires else range(op_cls.num_wires))
+        copied_op = copy.copy(op)
+        np.testing.assert_allclose(op.matrix(), copied_op.matrix(), atol=tol)
```

Comment:
Looking at `test_non_parametric_ops.py` now, I see this is how we were indeed testing the `copy` method.  Probably not the best way to do it.  

Since this is just moving pre-existing code, I won't complain too much, but I would prefer if we actually tested the attributes that needed to be copied, like `data`, `_wires`, and `_hyperparameters`.

Reply:
Maybe we could use `qml.equal` here too? If I understand it correctly, it'll check that these attributes are the same between two operators.